src/openfermion/transforms/_conversion_test.py

```diff
@@ -507,3 +517,490 @@ def test_get_fermion_operator_majorana_operator():
 def test_get_fermion_operator_wrong_type():
     with pytest.raises(TypeError):
         _ = get_fermion_operator(QubitOperator())
+
+
+class GetNumberPreservingSparseOperatorIntegrationTestLiH(unittest.TestCase):
+
+    def setUp(self):
+        # Set up molecule.
+        geometry = [('Li', (0., 0., 0.)), ('H', (0., 0., 1.45))]
+        basis = 'sto-3g'
+        multiplicity = 1
+        filename = os.path.join(THIS_DIRECTORY, 'data',
+                                'H1-Li1_sto-3g_singlet_1.45')
+        self.molecule = MolecularData(
+            geometry, basis, multiplicity, filename=filename)
+        self.molecule.load()
+
+        # Get molecular Hamiltonian.
+        self.molecular_hamiltonian = self.molecule.get_molecular_hamiltonian()
+
+        self.hubbard_hamiltonian = fermi_hubbard(
+            2, 2, 1.0, 4.0,
+            chemical_potential=.2,
+            magnetic_field=0.0,
+            spinless=False)
+
+
+    def test_number_on_reference(self):
+        sum_n_op = FermionOperator()
+        sum_sparse_n_op = get_number_preserving_sparse_operator(
+            sum_n_op,
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=False)
+
+        space_size = sum_sparse_n_op.shape[0]
+        reference = numpy.zeros((space_size))
+        reference[0] = 1.0
+
+        for i in range(self.molecule.n_qubits):
+            n_op = FermionOperator(((i, 1), (i, 0)))
+            sum_n_op += n_op
+
+            sparse_n_op = get_number_preserving_sparse_operator(
+                n_op,
+                self.molecule.n_qubits,
+                self.molecule.n_electrons,
+                spin_preserving=False)
+
+            sum_sparse_n_op += sparse_n_op
+
+            expectation = reference.dot(sparse_n_op.dot(reference))
+
+            if i < self.molecule.n_electrons:
+                assert expectation == 1.0
+            else:
+                assert expectation == 0.0
+
+        convert_after_adding = get_number_preserving_sparse_operator(
+            sum_n_op,
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=False)
+
+        assert scipy.sparse.linalg.norm(convert_after_adding - sum_sparse_n_op) < 1E-9
+
+        assert reference.dot(sum_sparse_n_op.dot(reference)) - self.molecule.n_electrons < 1E-9
+
+
+    def test_hf_energy(self):
+        hamiltonian_fop = get_fermion_operator(self.molecular_hamiltonian)
+
+        sparse_ham = get_number_preserving_sparse_operator(
+            hamiltonian_fop,
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=True)
+
+        space_size = sparse_ham.shape[0]
```

Comment:
Add a test that the resulting space size is what you'd expect (i.e. much smaller than a non-number-preserving operator).

Reply:
Done. I specifically check that space_size is what should be expected (and this number is smaller than the naive size of course).